From 9668848092d57b961480424853626b144f2e132c Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Thu, 29 Jul 2021 06:40:37 +0300
Subject: [PATCH] [SW-45031] greco: avoid onix specific duplications

In this patch, Greco HL_TARGET onix specific
common code moved to common dir.

Change-Id: Id77de4989fbfda66ac47d2429c9cee0fd69f221d
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/greco/arc/onix.c b/arch/greco/arc/onix.c
index d3e55fd..d9992e6 100644
--- a/arch/greco/arc/onix.c
+++ b/arch/greco/arc/onix.c
@@ -1,39 +1,13 @@
+// SPDX-License-Identifier: GPL-2.0+
 /*
  * (C) Copyright 2019 HabanaLabs, Ltd.
  * Noam Camus <noamikong@gmail.com>
  *
- * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <stdbool.h>
 
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-#include <cmd_module.h>
 
-#include <psoc_global_conf_regs.h>
-
-#define SYS_CLOCK_FREQUENCY	35000000ULL
-#define UART_CLOCK_FREQUENCY	SYS_CLOCK_FREQUENCY
-#define PLDM_UART_BAUDRATE	57600
-
-u32 hl_skip_pcie_init(void)
-{
-	return 1;
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
-void hl_uart_init(void)
-{
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
-}
 
 u32 hl_spi_baudr_get(void)
 {
@@ -43,16 +17,4 @@
 	return (factor + 1) & 0xFFFE;
 }
 
-enum cmd_check_res dev_specific_module_pci_check(void)
-{
-	/*
-	 * FLR and Hot Reset Registers are not part of PSOC and hence
-	 * writing to them results in ERROR that is resetting the
-	 * ONIX. To avoid it, simply return with NOT FOUND
-	 */
-	return NOT_FOUND;
-}
-
 void preboot_init_pcie_phy(void) {}
-
-void hl_dev_ts_init(void) {}
diff --git a/arch/greco/arc_elf32/onix.c b/arch/greco/arc_elf32/onix.c
index 110e7e4..7d65917 100644
--- a/arch/greco/arc_elf32/onix.c
+++ b/arch/greco/arc_elf32/onix.c
@@ -5,40 +5,18 @@
  */
 
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
 #include <efuse.h>
 
-#include <psoc_global_conf_regs.h>
 
 #include "common.h"
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
-#define UART_CLOCK_FREQUENCY	SYS_CLOCK_FREQUENCY
-#define PLDM_UART_BAUDRATE	57600
-
 #define EFUSE_SYSCFG_ONIX_DEFAULT \
 				(EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
 				 EFUSE_SYSCFG_KEY_LOCATION_SPI | \
 				 EFUSE_SYSCFG_REVOKATION_LOCATION_SPI | \
 				 EFUSE_SYSCFG_ENABLE_VALIDATION_MASK)
 
-u32 hl_skip_pcie_init(void)
-{
-	return 1;
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
-void hl_uart_init(void)
-{
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
-}
 
 u32 hl_spi_baudr_get(void)
 {
diff --git a/arch/greco/arm/onix.c b/arch/greco/arm/onix.c
index 1d93e28..ecfd98d 100644
--- a/arch/greco/arm/onix.c
+++ b/arch/greco/arm/onix.c
@@ -1,33 +1,15 @@
+// SPDX-License-Identifier: GPL-2.0+
 /*
  * (C) Copyright 2019 HabanaLabs, Ltd.
  * Noam Camus <noamikong@gmail.com>
  *
- * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <stdbool.h>
 
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-#include <cmd_module.h>
+
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
-
-u32 hl_skip_pcie_init(void)
-{
-	return 1;
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
-void hl_uart_init(void)
-{
-	preboot_uart_init(SYS_CLOCK_FREQUENCY / (HL_UART_BAUDRATE / 2));
-}
 
 u32 hl_spi_baudr_get(void)
 {
@@ -38,15 +20,3 @@
 	/* baud rate doesn't support odd number */
 	return (factor + 1) & 0xFFFE;
 }
-
-enum cmd_check_res dev_specific_module_pci_check(void)
-{
-	/*
-	 * FLR and Hot Reset Registers are not part of PSOC and hence
-	 * writing to them results in ERROR that is resetting the
-	 * ONIX. To avoid it, simply return with NOT FOUND
-	 */
-	return NOT_FOUND;
-}
-
-void hl_dev_ts_init(void) {}
diff --git a/arch/greco/common/onix.c b/arch/greco/common/onix.c
index b428f5a..bb5eae7 100644
--- a/arch/greco/common/onix.c
+++ b/arch/greco/common/onix.c
@@ -5,4 +5,39 @@
  *
  */
 
-/* TODO: Greco onix(HL_TARGET) specific common functions to be moved here */
+#include <preboot.h>
+#include <cmd_module.h>
+#include <psoc_global_conf_regs.h>
+#include <uart.h>
+
+#define SYS_CLOCK_FREQUENCY	35000000ULL
+
+u32 hl_skip_pcie_init(void)
+{
+	return 1;
+}
+
+u64 hl_pll_timestamp_freq_get(void)
+{
+	return SYS_CLOCK_FREQUENCY;
+}
+
+void hl_dev_ts_init(void) {}
+
+enum cmd_check_res dev_specific_module_pci_check(void)
+{
+	/*
+	 * FLR and Hot Reset Registers are not part of PSOC and hence
+	 * writing to them results in ERROR that is resetting the
+	 * ONIX. To avoid it, simply return with NOT FOUND
+	 */
+	return NOT_FOUND;
+}
+
+void hl_uart_init(void)
+{
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
+	preboot_uart_init(SYS_CLOCK_FREQUENCY / (HL_UART_BAUDRATE / 2));
+}
+
